primitives/runtime/src/offchain/rpc.rs

```diff
@@ -0,0 +1,431 @@
+		Request::default()
+	}
+
+	/// Change the Method for the Request
+	pub fn method(mut self, method: &'a str) -> Self {
+		self.method = method;
+		self
+	}
+
+	/// Change the Params for the Request
+	pub fn params(mut self, params: Vec<&'a str>) -> Self {
+		self.params = params;
+		self
+	}
+
+	/// Change the Jsonrpc for the Request
+	pub fn jsonrpc(mut self, jsonrpc: &'a str) -> Self {
+		self.jsonrpc = jsonrpc;
+		self
+	}
+
+	/// Change the ID for the Request
+	pub fn id(mut self, id: u32) -> Self {
+		self.id = id;
+		self
+	}
+}
+
+impl<'a> Client<'a> {
+	/// Create a new Client
+	pub fn new() -> Self {
+		Client::default()
+	}
+
+  /// Change the URL for the Client
+	pub fn url(mut self, url: &'a str) -> Self {
+		self.url = url;
+		self
+	}
+
+  /// Change the Timeout for the Client
+	pub fn timeout(mut self, timeout: u64) -> Self {
+		self.timeout = timeout;
+		self
+	}
+
+  /// Send the request and return a RPC result.
+	///
+	/// Err is returned in case there is a Http
+	/// or deserializing error.
+	pub fn send(&self, request: &'a Request) -> RpcResult {
+		let deadline = sp_io::offchain::timestamp().add(Duration::from_millis(self.timeout));
+		self.send_with_deadline(deadline, request)
+	}
+
+	/// Send the request and return a RPC result.
+	/// Deadline is used instead of Timeout
+	/// Err is returned in case there is a Http
+	/// or deserializing error.
+	pub fn send_with_deadline(&self, deadline: Timestamp, request: &'a Request) -> RpcResult {
+		// Construct http POST body
+		let request_body = json!({
+			"jsonrpc": request.jsonrpc,
+			"id": request.id,
+			"method": request.method,
+			"params": request.params
+		});
+		let mut body: Vec<&[u8]> = Vec::new();
+		let request_body_slice: &[u8] = &(serde_json::to_vec(&request_body).unwrap())[..];
+		body.push(request_body_slice);
+
+		// Send http POST request
+		let post_request = http::Request::post(self.url, body);
+
+		let pending = match post_request
+			.add_header("Content-Type", "application/json;charset=utf-8")
+			.deadline(deadline)
+			.send().map_err(|_| http::Error::IoError) {
+				Ok(result) => result,
+				Err(e) => return Err(Error::Http(e)),
+		};
+
+		// Hanlde http response
+		let response = match pending.try_wait(deadline)
+			.map_err(|_| http::Error::DeadlineReached) {
+				Ok(result) =>
+					match result {
+						Ok(result_bis) => result_bis,
+						Err(e) => return Err(Error::Http(e)),
+					},
+				Err(e) => return Err(Error::Http(e)),
+		};
+
+		// Return http error if response status is not 200
+		if response.code != 200 {
+			log::warn!("Unexpected status code: {}", response.code);
+			return Err(Error::Http(http::Error::Unknown));
+		}
+
+		// Deserialize response body
+		let response_body_bytes = response.body().collect::<Vec<u8>>();
+		let rpc_result = Self::deserialize_response(&response_body_bytes)?;
+
+		Ok(rpc_result)
+	}
+
+	/// Deserialize a RPC response
+	pub fn deserialize_response(response_body_bytes: &Vec<u8>) -> RpcResult {
```

Comment:
> Does it need to be public?

Not sure if it can be useful to reuse for the upcoming internal RPC calls?

I changed it to a private standalone function.